service/controller/v17/resource/endpoint/delete.go

```diff
@@ -24,7 +25,7 @@ func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange inte
 	// times. We do not want to delete the whole endpoint only because one pod is
 	// gone. We only delete the whole endpoint when it does not contain any IP
 	// anymore. Removing IPs is done on update events.
-	if isEmptyEndpoint(*endpointToDelete) {
+	if endpointToDelete != nil && isEmptyEndpoint(endpointToDelete) {
```

Comment:
I don't see a need to check for nil pointer here since this check is already done in `isEmptyEndpoint()`

Reply:
I did not see it either in the first place. Then I noticed that the handling between create/update and delete is different. The endpoint can be nil, or not nil and empty regardless. `isEmptyEndpoint` returns true when the endpoint is nil. When the endpoint is nil, we do not want to delete. We want to delete though in case the endpoint is empty. So we have to check this before. Alternatively we could remove the nil check from the method and test it also in create and update more explicitly. What do you think?